### PR TITLE
Puppet lint, style guide, and spec fixes

### PR DESCRIPTION
**Pull request by jeefberkey**

### CODE REVIEW

**Comment by ripienaar** [comment]
This looks good except the change to single quotes from double :)

**Comment by jeefberkey** [comment]
I can put them back and add a parameter in `.puppet-lint.rc`, one minute

**Comment by ripienaar** [comment]
Kewl. It's easier to stay consistent with doubles throughout than singles. Many other languages are starting to adjust their style guides to favour double quotes and for me the choice is quite deliberate to use them. Appreciate it!

File: manifests/init.pp [hunk 1]
```diff
@@ -37,7 +63,8 @@
     if $manage_collectd {
       include nats::collectd
     }
-  } else {
+  }
```
Comment (ripienaar) [inline manifests/init.pp]: this should be on the same line as before

File: manifests/init.pp [hunk 2]
```diff
@@ -1,3 +1,29 @@
...
+# @param binary_source
+# @param service_name
+# @param service_ensure
+# @param service_type
+# @param cert_file
+# @param key_file
+# @param ca_file
+# @param manage_collectd
+# @param limit_nofile
+#
```
Comment (ripienaar) [inline manifests/init.pp]: YARD generally does not like blank comments at the end, not sure if its the same with puppet but I'd prefer this was not here.  

Generally I'd question the value of these comments without details or types etc, puppet strings will automatically pull out the variables and types and everything for you, the thing it does on its own is more detailed than these comments so I doubt its needed.

Here's a example of it without the comment section https://devco.net/rip/puppet-nats/puppet_classes/nats.html

File: spec/classes/init_spec.rb [hunk 3]
```diff
@@ -8,7 +8,8 @@
       "networking" => {
         "fqdn" => "rspec.example.com"
       },
-      "PATH" => '/usr/bin'
+      "PATH" => '/usr/bin',
+      'operatingsystemmajrelease' => '6'
```
Comment (ripienaar) [inline spec/classes/init_spec.rb]: mind making these `"` as well?

File: manifests/init.pp (hunk 2 above)
Comment (jeefberkey) [inline manifests/init.pp]: This is a struggle I've been having myself. I'm not spun up on NATS enough to actually write the docs, so I'm going to agree with you.

File: spec/classes/init_spec.rb (hunk 3 above)
Comment (jeefberkey) [inline spec/classes/init_spec.rb]: I went ahead and did the rest of them in that file as well.

**Comment by ripienaar** [comment]
looking great, do you mind squashing down to a single commit @jeefberkey ?

**Comment by jeefberkey** [comment]
@ripienaar np, I'm used to the squash and merge workflow

**Comment by ripienaar** [comment]
thanks a lot @jeefberkey !

**Pull request merged**